WebProcessProxy should use default settings values for near-suspended assertions when they have no page

cdumez opened the pull request:
#### a6ed791b6aab1c26bfeeb81eb353d777a4282cc0
<pre>
WebProcessProxy should use default settings values for near-suspended assertions when they have no page
<a href="https://bugs.webkit.org/show_bug.cgi?id=256648">https://bugs.webkit.org/show_bug.cgi?id=256648</a>
rdar://108292314

Reviewed by Sihui Liu.

WebProcessProxy should use default settings values for near-suspended
assertions when they have no page. In WebKit, settings come from pages. The
WebProcessProxy therefore iterate through pages to figure out if it should take
near-suspended assertions and if it should drop them after a delay.

However, if the process doesn&apos;t have any pages, it would fall back to a value
that is not the default value for these settings, which would lead to
unexpected behavior in some case. This is particularly noticeable for service
worker processes which often don&apos;t have pages.

* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::shouldTakeNearSuspendedAssertion const):
(WebKit::WebProcessProxy::shouldDropNearSuspendedAssertionAfterDelay const):

Canonical link: <a href="https://commits.webkit.org/263975@main">https://commits.webkit.org/263975@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1ee31d4cb8748cc64816eeb61603c4f04f317bb0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/6262 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/6457 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/6636 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/7835 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/6592 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/6677 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/6407 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/9489 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/6374 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/6388 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/5673 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/7905 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/3860 "Passed tests") | | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/13543 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/5724 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/5735 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/7982 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/6212 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/5083 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/5624 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1488 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/9777 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/5992 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

szewai reviewed (approved):
r=me
It seems these settings should be on WebProcessPool though

cdumez committed:
WebProcessProxy should use default settings values for near-suspended assertions when they have no page
https://bugs.webkit.org/show_bug.cgi?id=256648
rdar://108292314

Reviewed by Sihui Liu.

WebProcessProxy should use default settings values for near-suspended
assertions when they have no page. In WebKit, settings come from pages. The
WebProcessProxy therefore iterate through pages to figure out if it should take
near-suspended assertions and if it should drop them after a delay.

However, if the process doesn't have any pages, it would fall back to a value
that is not the default value for these settings, which would lead to
unexpected behavior in some case. This is particularly noticeable for service
worker processes which often don't have pages.

* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::shouldTakeNearSuspendedAssertion const):
(WebKit::WebProcessProxy::shouldDropNearSuspendedAssertionAfterDelay const):

Canonical link: https://commits.webkit.org/263975@main
